### PR TITLE
Experimental/mapper update

### PR DESCRIPTION
**Pull request by avullo**
This is to include work done on improving search for overlapping intervals using Interval Trees (ENSCORESW-2337) and updating the Slice project/project_to_slice methods, and correspondingly the higher level mappers, to correctly return the coordinates of the projected bits of the source slice (ENSCORESW-2289).

### CODE REVIEW

File: modules/Bio/EnsEMBL/AssemblyMapper.pm [hunk 1]
```diff
@@ -223,7 +223,7 @@ sub map {
   throw('Incorrect number of arguments.') if (!( @_ >= 6));
 
   my ( $self, $frm_seq_region_name, $frm_start, $frm_end, $frm_strand,
-       $frm_cs, $to_slice )
+       $frm_cs, $dummy, $to_slice, $include_org_coord )
```
Comment (lairdm) [inline modules/Bio/EnsEMBL/AssemblyMapper.pm]: The documentation doesn't match the new function prototype, for example what this $dummy parameter is.

File: modules/Bio/EnsEMBL/ChainedAssemblyMapper.pm [hunk 2]
```diff
@@ -276,7 +276,7 @@ sub map {
   throw('Incorrect number of arguments.') if(@_ < 6);
 
   my ($self, $frm_seq_region_name, $frm_start,
-      $frm_end, $frm_strand, $frm_cs, $fastmap, $to_slice) = @_;
+      $frm_end, $frm_strand, $frm_cs, $fastmap, $to_slice, $include_org_coord) = @_;
```
Comment (lairdm) [inline modules/Bio/EnsEMBL/ChainedAssemblyMapper.pm]: Again, sorry, missing this new 8th argument flag description.

Comment (avullo) [reply modules/Bio/EnsEMBL/ChainedAssemblyMapper.pm]: You're perfectly right and I was aware of this inconsistency, thanks for the reminder! :+1:

**Comment by magaliruffier** [comment]
Some changes pending:

- review the description of arguments for methods where the order of arguments has changed
- rebase against master, we should not be getting all the copyright updates
- ensure xref code returns comparable results with new tree implementation, see https://www.ebi.ac.uk/panda/jira/browse/ENSCORESW-2602?filter=14252

once this has been addressed, we can review the merge conflicts

**Comment by nerdstrike** [comment]
@avullo What is the state of this pull-request? Are you happy that you've resolved all comments so far?

**Comment by avullo** [comment]
I've got to roll-back wrt some later updates concerning the RangeRegistry and re implement differently. Then I've got to address one last issue which has appeared recently.  I've discussed with Mag and we think it's better if I close this and make another one.

**Comment by nerdstrike** [comment]
Thanks @avullo ! Closing - the trail of critique has gone cold.

**Pull request closed without merging**